Nodes/xBeeSyncNode.h

```diff
@@ -0,0 +1,87 @@
+
+/****************************************************************************************
+ *
+ * File:
+ * 		xBeeSyncNode.h
+ *
+ * Purpose:
+ *		Handles sending of logs or the current system state (xml-format) over xbee,
+ * 		depending on the setting of send_logs. Also receives rudder and sail control messages
+ *		over the xbee network.
+ *
+ * Developer Notes:
+ *
+ *
+ ***************************************************************************************/
+
+ #pragma once
+
+#include "ActiveNode.h"
+#include "xBee/xBee.h"
+#include "xmlparser/src/xml_log.h"
+#include "logger/Logger.h"
+#include "dbhandler/DBHandler.h"
+#include "utility/Timer.h"
+#include "Messages/VesselStateMsg.h"
+#include "Messages/ActuatorPositionMsg.h"
+
+class xBeeSyncNode : public ActiveNode {
+public:
+	xBeeSyncNode(MessageBus& msgBus, DBHandler* db);
+
+	///----------------------------------------------------------------------------------
+	/// Gets all settings from database and initializes xbee object
+	///
+	///----------------------------------------------------------------------------------
+	bool init();
+
+	///----------------------------------------------------------------------------------
+	/// Starts the thread if init is done
+	///
+	///----------------------------------------------------------------------------------
+	void start();
+	void processMessage(const Message*);
+	
+private:
+
+	///----------------------------------------------------------------------------------
+	/// Sends an xml-formated vessel state string over the xbee network
+	///
+	///----------------------------------------------------------------------------------
+	void sendVesselState(VesselStateMsg* msg);
+
+	///----------------------------------------------------------------------------------
+	/// Sends a log string over the xbee network
+	///
+	///----------------------------------------------------------------------------------
+	void sendLogs();
+
+	///----------------------------------------------------------------------------------
+	/// Generates an actuator control message from data received by xbee
+	///
+	///----------------------------------------------------------------------------------
+	void receiveControl();
+
+	///----------------------------------------------------------------------------------
+	/// Polls receiveControl and sendlogs repeatedly
+	///
+	///----------------------------------------------------------------------------------
+	static void xBeeSyncThread(void* nodePtr);
+
+	XML_log m_XML_log;
```

Comment:
This isn't really a fix comment but a more general question. Why do we use XML over radio comms, why not compress the data then turn it to xml on the xbee relay side?